rkt/api_service.go

```diff
@@ -305,76 +343,194 @@ func getNetworks(p *pod) []*v1alpha.Network {
+				Type:    v1alpha.ImageType_IMAGE_TYPE_APPC,
+				Version: schema.AppContainerVersion.String(),
+			},
+			Id: fullImageID,
+			// Other information are not available because they require the image
+			// info from store. Some of it is filled in below if possible.
+		}
+
+		im, err := p.getAppImageManifest(*types.MustACName(app.Name))
+		if err != nil {
+			stderr.PrintE(fmt.Sprintf("failed to get image manifests for app %q", app.Name), err)
+			errlist = append(errlist, err)
+		} else {
+			app.Image.Name = im.Name.String()
+
+			version, ok := im.Labels.Get("version")
+			if !ok {
+				version = "latest"
+			}
+			app.Image.Version = version
+		}
+	}
+
+	if len(errlist) != 0 {
+		return errs{errlist}
+	}
+	return nil
+}
+
+func (s *v1AlphaAPIServer) getBasicPodFromDisk(p *pod) (*v1alpha.Pod, error) {
+	pod := &v1alpha.Pod{Id: p.uuid.String()}
 
-	manifest, data, err := getPodManifest(p)
+	manifest, data, err := s.getPodManifest(p)
 	if err != nil {
 		stderr.PrintE(fmt.Sprintf("failed to get the pod manifest for pod %q", p.uuid), err)
 	} else {
-		pod.Manifest = data
 		pod.Annotations = convertAnnotationsToKeyValue(manifest.Annotations)
 		pod.Apps = getApplist(manifest)
+		pod.Manifest = data
+		err = fillStaticAppInfo(s.store, p, pod)
+	}
+
+	return pod, err
+}
+
+// getBasicPod returns v1alpha.Pod with basic pod information.
+func (s *v1AlphaAPIServer) getBasicPod(p *pod) *v1alpha.Pod {
+	mtime, mtimeErr := p.getModTime("pod")
+	if mtimeErr != nil {
+		stderr.PrintE(fmt.Sprintf("failed to read the pod manifest's mtime for pod %q", p.uuid), mtimeErr)
 	}
 
+	// Couldn't use pod.uuid directly as it's a pointer.
+	itemValue, found := s.podCache.Get(p.uuid.String())
+	if found && mtimeErr == nil {
+		cacheItem := itemValue.(*podCacheItem)
+
+		// Check the mtime to make sure we are not returning stale manifests.
+		if !mtime.After(cacheItem.mtime) {
+			return copyPod(cacheItem.pod)
+		}
+	}
+
+	pod, err := s.getBasicPodFromDisk(p)
+	if mtimeErr != nil || err != nil {
+		// If any error happens or the mtime is unknown,
+		// returns the raw pod directly without adding it to the cache.
+		return pod
+	}
+
+	cacheItem := &podCacheItem{pod, mtime}
+	s.podCache.Add(p.uuid.String(), cacheItem)
+
+	// Return a copy of the pod, so the cached pod is not mutated later.
+	return copyPod(cacheItem.pod)
+}
+
+func waitForMachinedRegistration(uuid string) error {
+	conn, err := dbus.SystemBus()
+	if err != nil {
+		return err
+	}
+	machined := conn.Object("org.freedesktop.machine1", "/org/freedesktop/machine1")
+	machineName := "rkt-" + uuid
+
+	var o dbus.ObjectPath
+	for i := 0; i < 10; i++ {
+		if err := machined.Call("org.freedesktop.machine1.Manager.GetMachine", 0, machineName).Store(&o); err == nil {
+			return nil
+		}
+		time.Sleep(time.Millisecond * 50)
+	}
+
+	return errors.New("pod not found")
+}
+
+// fillPodDetails fills the pod's dynamic info, e.g. the pod's state,
+// the pod's network info, the apps' state, etc.
```

Comment:
`pod` is a bit ambiguous in this context, I think it should mention explicitly that `v1pod` is the target of this filling.